pkg/didcomm/protocol/introduce/service.go

```diff
@@ -275,14 +275,19 @@ func (s *Service) InvitationReceived(msg service.StateMsg) error {
 		return nil
 	}
 
-	if msg.Msg.Header == nil || msg.Msg.Header.Thread.PID == "" {
+	var h = service.Header{}
+	if err := msg.Msg.ToStruct(&h); err != nil {
```

Comment:
I really think `Scan` would be a better name. `ToStruct` implies the whole message will be scanned into the input struct.